src/app/features/vendor/vendor-view/vendor-view.component.ts

```diff
@@ -0,0 +1,96 @@
+import { Component } from '@angular/core';
+import { ConfirmationService } from 'primeng/api';
+import { IVendor, VendorService } from '../services/vendor.service';
+import { Router } from '@angular/router';
+
+/**  LLD
+
+ * This component is for viewing a particular vendor's details
+
+ * It displays the following information:
+ * In the header section,Displays the name of the vendor and if the vendor is not yet approved a pending status label will be shown besides the name .
+ * Includes action buttons for "Edit Profile" and "Approve Vendor."
+ * 
+ * In the content section the folowing details are displayed
+ * email
+ * phone number
+ * website
+ * market
+ * services
+ * 
+ * It has the following buttons:
+ * Edit Profile
+ * Approve Vendor
+
+ * This screen contains the following actions:
+ * Display the specific vendor's details
+ * Navigate to the edit vendor profile page when clicked on the Edit profile button.
+ * Approve the vendor's status from pending to approved when the user clicks the Approve Vendor button. 
+ */
+
+@Component({
+  selector: 'r2q-vendor-view',
+  standalone: false,
+  templateUrl: './vendor-view.component.html',
+  styleUrl: './vendor-view.component.scss',
+  providers:[ConfirmationService]
+})
+
+export class VendorViewComponent {
+  vendor!: IVendor;
+  constructor(
+    private confirmationService: ConfirmationService,
+    private vendorService: VendorService,
+    private router: Router
+  ) {}
+
+  /** 
+   * On component initialization, fetches the vendor details
+   */
+  ngOnInit(): void {
+    this.getVendorById(2);
```

Comment:
Why id is hardcoded here? cant we get id from route?

Reply:
Since we haven't integrated all the modules yet I am using hardcoded values to do the work. Will change that after integration.

Reply:
Changed to getting id from route.